webapp/tests/karma/unit/services/telemetry.js

```diff
@@ -0,0 +1,215 @@
+describe('Telemetry service', () => {
+  'use strict';
+
+  const NOW = new Date(2018, 10, 10, 12, 33).getTime();
+
+  let clock,
+    service,
+    $window,
+    DB,
+    storageGetItem,
+    storageSetItem,
+    storageRemoveItem,
+    pouchDb;
+
+  beforeEach(() => {
+    module('inboxApp');
+    clock = sinon.useFakeTimers(NOW);
+
+    storageGetItem = sinon.stub();
+    storageSetItem = sinon.stub();
+    storageRemoveItem = sinon.stub();
+
+    pouchDb = {};
```

Comment:
I'd be tempted to use an in-memory pouchdb here.

Reply:
I'd like to keep unit tests as fast as possible so I think we should stick with mocking here. There would be some value in adding integration tests which could use a real pouchdb to ensure the units work together as expected.